go/aead/aead_factory.go

```diff
@@ -47,24 +48,34 @@ type wrappedAead struct {
 // Asserts that wrappedAead implements the AEAD interface.
 var _ tink.AEAD = (*wrappedAead)(nil)
 
-func newWrappedAead(ps *primitiveset.PrimitiveSet) *wrappedAead {
+func newWrappedAead(ps *primitiveset.PrimitiveSet) (*wrappedAead, error) {
+	if _, ok := (ps.Primary.Primitive).(tink.AEAD); !ok {
```

Comment:
We should check all primitives in the keyset; they all need to be convertible to AEAD objects (otherwise, a keyset with mixed keys will silently behave weirdly, which is not the desired behavior).

Hence, we should check whether all primitives can be cast to AEAD here.

Reply:
Sure, I will add the check on all primitives in the keyset.

Reply:
done